src/jcli/file/files-man.ts

```diff
@@ -0,0 +1,78 @@
+import { extname, join } from "path";
+import { api } from "@/api/mod.ts";
+import { digest } from "@/jcli/crypto.ts";
+
+export class FileEntry {
+  #digest: string | undefined;
+
+  constructor(public readonly path: string) {}
+
+  async digest(): Promise<string> {
+    if (undefined === this.#digest) {
+      const bytes = await api.fs.readFile(this.path);
+      this.#digest = await digest(bytes);
+    }
+
+    return this.#digest;
+  }
+}
+
+export interface FileCreated {
+  type: "CREATED";
+  path: string;
+  digest: string;
+}
+
+export interface FileUpdated {
+  type: "UPDATED";
+  path: string;
+  digest: string;
+}
+
+export interface FileDeleted {
+  type: "DELETED";
+  path: string;
+}
+
+export type FileChange = FileCreated | FileUpdated | FileDeleted;
+
+function hasExtension(
+  path: string,
+  extension: ReadonlyArray<string> | string,
+): boolean {
+  const ext = extname(path);
+
+  if (typeof extension === "string") {
+    return ext === extension;
+  } else {
+    return extension.some((e) => ext === e);
+  }
+}
+
+export async function* listFiles(
+  path: string,
+  extension?: ReadonlyArray<string> | string,
+): AsyncIterable<FileEntry> {
+  for await (const entry of api.fs.readDir(path)) {
+    if (entry.isFile) {
+      if (undefined === extension || hasExtension(entry.name, extension)) {
+        yield new FileEntry(join(path, entry.name));
+      }
+    }
+  }
```

Comment:
提供一个思路，创建一个临时函数，可以增强可读性和代码组织

```suggestion
  const filter = (entry) => {
    if (!entry.isFile) return false;
    // keep the file if the extension is not given
    if (undefined === extension) return true;

    return hasExtension(entry.name, extension);
  }
  for await (const entry of api.fs.readDir(path)) {
    filter(entry) && yield new FileEntry(join(path, entry.name));
  }
```